csfieldguide/static/interactives/lzw-compression/js/lzw-compression.js

```diff
@@ -0,0 +1,265 @@
+            });
+            code_div.addEventListener('mouseleave', function(event) {
+                changeHighlight(event, false);
+            });
+            code_div.addEventListener('focus', function(event) {
+                changeHighlight(event, true);
+            });
+            code_div.addEventListener('focusout', function(event) {
+                changeHighlight(event, false);
+            });
+
+            line_div.appendChild(code_div);
+        }
+    }
+    encoded_message_fragment.appendChild(line_div);
+    encoded_message_div.appendChild(encoded_message_fragment);
+}
+
+
+// highlights the corresponding dictionary entry
+function changeHighlight(event, highlight) {
+    var selected_reference;
+    if (event.nodeName == 'INPUT') { // event was changing focus to next input box
+        selected_reference = event.parentElement;
+    } else {
+        selected_reference = event.target;
+        if (event.target.nodeName == 'P' || event.target.nodeName == 'INPUT') {
+            selected_reference = event.srcElement.parentElement;
+        }
+    }
+    var code_index = parseInt(selected_reference.dataset.referenceIndex);
+    var code_div = document.querySelectorAll('[data-index="' + code_index.toString() + '"]')[0];
+    var code_div_top_position = code_div.offsetTop;
+    
+    if (highlight == true) {
+        code_div.classList.add('highlight');
+        selected_reference.classList.add('selected');
+        document.getElementById('interactive-lzw-compression-dictionary').scrollTop = code_div_top_position - 60;
+    } else {
+        code_div.classList.remove('highlight');
+        selected_reference.classList.remove('selected');
+    }
+}
+
+// add each of the dictionary entries to the page
+function makeDictionaryEntryElements(start_index, end_index, element) {
+    var fragment = document.createDocumentFragment();
+    for (var i = start_index; i < end_index; i++) {
+        // create dictionary entry element
+        var dictionary_entry = document.createElement('div');
+        dictionary_entry.classList.add('interactive-lzw-dictionary-entry');
+        dictionary_entry.setAttribute('data-index', i);
+        
+        // create code element
+        var code_element = document.createElement('p');
+        code_element.innerHTML = i;
+        code_element.classList.add('interactive-lzw-dictionary-code');
+        
+        // create value element
+        var value_element = document.createElement('p');
+        value_element.innerHTML = codes[i].replace(/\s+/g, '_');
+        value_element.classList.add('interactive-lzw-dictionary-value');
+        
+        // add code and value to parent div
+        dictionary_entry.appendChild(code_element);
+        dictionary_entry.appendChild(value_element);
+        fragment.append(dictionary_entry);
+    }
+    element.appendChild(fragment);
+};
+
+
+// display the dictionary built by LZW, in 3 columns
+function outputDictionary() {
```

Comment:
If a dictionary is displayed, and I set the 'message to compress' to an empty string and hit 'compress' the dictionary doesn't change.
![image](https://user-images.githubusercontent.com/16066822/55924086-820ac000-5c5c-11e9-81ef-0174fc0bea5b.png)